packages/playwright-core/src/server/chromium/chromium.ts

```diff
@@ -63,6 +63,11 @@ export class Chromium extends BrowserType {
     if (options.headers)
       headersMap = headersArrayToObject(options.headers, false);
 
+    if (!headersMap)
```

Comment:
what if the header name is lower-case?

Reply:
headersMap ??= {}; 
if (!headersMap['User-Agent'])
  headersMap['User-Agent'] = getUserAgent();